withPrintFn (\_ -> return()) will now avoid writing to stdout

unhammer opened the pull request:
Fixes #2 - prints to stdout

Made the default print function use stderr, exposed
svm_set_print_string_function in SVM.Raw, and made a withPrintFn
helper to easily pass a Haskell print function in and free the
function pointer afterwards.

Also commented out the print_node(x) call from predict completely,
looked like a debug thing since it was commented out elsewhere …

paulrzcz commented:
@unhammer Hi, sorry for the delayed response. I've been a bit busy at my work recently.

I'm quite fine with your PR but I'd like to keep number of changes in libsvm as low as possible. Therefore, I'd prefer keep print_string_stdout function as it is in libsvm. Also changes in svm_set_print_string_function are not required in this case.

I'm fine to comment out print_node 'cause it is a bit noisy.

Are you fine to change your PR?

unhammer committed:
Can now do `withPrintFn (\_ -> return())` to avoid writing to stdout

Fixes #2 - prints to stdout

Exposed svm_set_print_string_function in SVM.Raw, and made a
withPrintFn helper to easily pass a Haskell print function in and free
the function pointer afterwards.

Also commented out the print_node(x) call from predict completely,
looked like a debug thing since it was commented out elsewhere …

unhammer commented:
Sure, that makes sense. I've rebased this PR with the changes; also added a little docstring to `withPrintFn` to make it easier on newcomers to turn off the output

paulrzcz commented:
Thanks a lot!

bgamari commented:
Can you make a release with this change? The `print_node` is indeed rather noisy.

paulrzcz commented:
The change has been just released as 0.1.1.3.22 including the one with `predict`.
